File: package/test/test_package.cpp
#include "test/test_package.hpp"

TEST(package_name, initialization)
{
    auto node = std::make_shared<PackageNode>();
    rclcpp::spin_some(node);
}

int main(int argc, char** argv)
{
    testing::InitGoogleTest(&argc, argv);
    rclcpp::init(argc, argv);
    auto result = RUN_ALL_TESTS();
    rclcpp::shutdown();
    return result;
}

File: package/test/test_package.py
import rclpy

try:
    from ..package.node import MinimalPublisher
except ImportError:
    from package.node import MinimalPublisher

def test_initialization():
    minimal_publisher = MinimalPublisher()

    assert True

def setup_module(module):
    rclpy.init()


def teardown_module(module):
    rclpy.shutdown()